docs/running/server/wes.rst

```diff
@@ -0,0 +1,272 @@
+
+As a quick example, we can submit the example CWL workflow from :ref:`cwlquickstart` to our WES API using cURL::
+
+    $ curl --location --request POST 'http://localhost:8080/ga4gh/wes/v1/runs' \
+        --form 'workflow_url="example.cwl"' \
+        --form 'workflow_type="cwl"' \
+        --form 'workflow_type_version="v1.0"' \
+        --form 'workflow_params="{\"message\": \"Hello world!\"}"' \
+        --form 'workflow_attachment=@"./toil_test_files/example.cwl"'
+    {
+      "run_id": "4deb8beb24894e9eb7c74b0f010305d1"
+    }
+
+
+If the workflow is submitted successfully, a JSON object containing a ``run_id`` will be returned. The ``run_id`` is a
+unique identifier of your requested workflow, which can be used to monitor or cancel the run.
+
+
+There are a few required parameters that have to be set for all workflow submissions, which are the following:
+
++---------------------------+-------------------------------------------------------------+
+| workflow_url              | The URL of the workflow to run. This can refer to a file    |
+|                           | from ``workflow_attachment``.                               |
++---------------------------+-------------------------------------------------------------+
+| workflow_type             | The type of workflow language. Toil currently supports one  |
+|                           | of the following: ``"CWL"``, ``"WDL"``, or ``"py"``. To run |
+|                           | a Toil script, set this to ``"py"``.                        |
++---------------------------+-------------------------------------------------------------+
+| workflow_type_version     | The version of the workflow language. Supported versions    |
+|                           | can be found by accessing the ``GET /service-info``         |
+|                           | endpoint of your WES server.                                |
++---------------------------+-------------------------------------------------------------+
+| workflow_params           | A JSON object that specifies the inputs of the workflow.    |
++---------------------------+-------------------------------------------------------------+
+
+Additionally, the following optional parameters are also available:
+
++--------------------------------+--------------------------------------------------------+
+| workflow_attachment            | A list of files associated with the workflow run.      |
++--------------------------------+--------------------------------------------------------+
+| workflow_engine_parameters     | A JSON key-value map of workflow engine parameters     |
+|                                | to send to the runner.                                 |
+|                                |                                                        |
+|                                | Example:                                               |
+|                                | ``{"--logLevel": "INFO", "--workDir": "/tmp/"}``       |
```

Comment:
Do we not ever need to be able to pass the same option multiple times (for e.g. `argparse` `nargs="+"` options)? Or do we not have any of those in Toil?

Reply:
We should definitely consider this case. However, the spec doesn't seem to specify the format of these parameters. All it says about `workflow_engine_parameters` is that it should be a [string](https://ga4gh.github.io/workflow-execution-service-schemas/docs/#operation/RunWorkflow).

We could probably make the structure the same as the [`default_workflow_engine_parameters`](https://github.com/ga4gh/workflow-execution-service-schemas/blob/c3b19854240c4fcbaf3483e22b19db0a918a7ee5/openapi/workflow_execution_service.swagger.yaml#L361-L372) to allow multiple options with the same name, but I feel like it'll be difficult to write by hand. (e.g.: `[{"name": "--logLevel", "value": "INFO"}, {"name": "--workDir", "value": "/tmp/"}]`).

Reply:
I think the key-value map is probably fine. We could also just use a shell-escaped string (`"--logLevel INFO --workDir /tmp/"`), but that might invite some unsafe parsing strategies.